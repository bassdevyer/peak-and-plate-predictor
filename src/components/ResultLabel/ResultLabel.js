import React from 'react'
import './style.css'
import { Label } from 'react-bootstrap'
import PropTypes from 'prop-types'

const propTypes = {
  style: PropTypes.string,
  label: PropTypes.string,
}

const ResultLabel = (props) => {
  const
    {
      style,
      label,
    } = props

  return (
    <Label bsStyle={style}>{label}</Label>
  )
}

ResultLabel.propTypes = propTypes

export default ResultLabel