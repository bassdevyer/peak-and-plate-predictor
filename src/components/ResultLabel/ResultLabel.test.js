import React from 'react'
import ResultLabel from '../ResultLabel'
import { create } from 'react-test-renderer'
import { mount } from 'enzyme/build'

describe('Component default rendering', () => {
  it('Renders component correctly', () => {
    const component = create(
      <ResultLabel/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Component props', () => {

  it('check if props are set correctly', () => {
    const props = {
        style: 'default',
        label: 'Complete los datos',
      },
      DateInputComponent = mount(<ResultLabel {...props} />)
    expect(DateInputComponent.prop('style'))
      .toEqual('default')
    expect(DateInputComponent.prop('label'))
      .toEqual('Complete los datos')

  })

  it('checks prop types', () => {
    const props = {
        style: 'default',
        label: 'Complete los datos',
      },
      DateInputComponent = mount(<ResultLabel {...props} />)
    expect(DateInputComponent.prop('style')).toBeString()
    expect(DateInputComponent.prop('label')).toBeString()
  })
})
