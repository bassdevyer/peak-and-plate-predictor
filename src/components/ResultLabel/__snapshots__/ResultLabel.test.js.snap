// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component default rendering Renders component correctly 1`] = `
<span
  className="label label-default hidden"
/>
`;
